ssc/cmod_pvwattsv8.cpp

```diff
@@ -826,12 +823,28 @@ class cm_pvwattsv8 : public compute_module
                 else if (is_assigned("albedo"))
                     log(util::format("Albedo array was assigned but is not the correct length (1, 12, or %d entries). Using default value.", nrec), SSC_WARNING);
 
-                // if the user hasn't specified an albedo, or has specified to use wf albedo, and the weather file contains hourly albedo, use that instead
-                // but make sure that the weather file albedo is realistic
-                // albedo_len will be zero if the albedo input isn't assigned
-                if (std::isfinite(wf.alb) && wf.alb > 0 && wf.alb < 1 && (albedo_len == 0 || use_wf_albedo))
+                // check albedo is valid value between 0 and 1. if not, set to default value 0.2
+                // if the snow loss model is enabled and there's valid snow depth > 0.5 cm, then set to 0.6
+                if (alb <= 0 || alb >= 1)
+                {
+                    if (std::isfinite(wf.snow) && wf.snow > 0.5 && wf.snow < 999 && en_snowloss)
+                        alb = as_double("albedo_default_snow");
+                    else
+                        alb = as_double("albedo_default");
+                }
+
+                // conditions to use albedo from wf instead of albedo user input:
+                //   * wf.alb exists and is valid and use_wf_albedo = 1
+                //   * wf.alb exists and is valid and use_wf_albedo = 0 and albedo_len = 0 
```

Comment:
Is this the desired behavior? You can remind me if we discussed at a meeting and I forgot, but I would think that if use_wf_albedo = 0 and user-specified albedo is not defined, I would be expecting it to use internal defaults (or the defaults that I set via the SDK), even if albedo exists in the weather file.

Reply:
I agree with the behavior you describe. Fixed in 22acecb6.

Reply:
Based on discussions at the 9/20/2022 SAM meeting, we agreed to simplify the albedo logic:

If use_wf_albedo is true, and wf albedo is available and valid, use wf albedo, otherwise use default albedo (snow or no snow, depending on snow depth and snow model enabled).

If use_wf_albedo is false, and user albedo is available and valid, use user albedo, otherwise use default albedo as above.